In [107]:
import pandas as pd
import numpy as np
import warnings
warnings.filterwarnings("ignore")



## **Q1. Creating a DataFrame:-**
**Create a Pandas DataFrame using a Python dictionary with the following columns:**

- Product

- Region

- Sales

- Quantity

Add at least 5 rows of sample data and display the DataFrame.

In [8]:
# Ans 1

data = {
    "Product": ["Laptop", "Mobile", "Tablet", "Headphones", "Smartwatch"],
    "Region": ["North", "South", "East", "West", "Central"],
    "Sales": [50000, 30000, 20000, 10000, 25000],
    "Quantity": [10, 15, 8, 20, 12]
}


In [9]:
# DataFrame
df1= pd.DataFrame(data)
df1

,Product,Region,Sales,Quantity
0,Laptop,North,50000,10
1,Mobile,South,30000,15
2,Tablet,East,20000,8
3,Headphones,West,10000,20
4,Smartwatch,Central,25000,12


## **Q2. Data Inspection:-**

**Perform the following inspection tasks:**

- Display the first 5 rows of the dataset.

- Display the last 5 rows.

- Show the structure of the dataset using info().

- Display the shape of the dataset.

In [10]:
# Ans 2
# Importing Data

df2 = pd.read_csv('/content/retail_dataset .csv')
df2

,Transaction ID,Customer ID,Product Category,Quantity,Price,Date,Region
0,T001,C015,Books,7.0,167.0,02-02-2023,North
1,T002,C004,Toys,4.0,374.0,17-02-2023,South
2,T003,C033,Furniture,8.0,1059.0,17-05-2024,South
3,T004,C047,Groceries,5.0,304.0,11-02-2024,South
4,T005,C022,Books,7.0,282.0,27-07-2024,West
...,...,...,...,...,...,...,...
95,T096,C036,Furniture,10.0,2901.0,08-08-2023,North
96,T097,C021,Clothing,9.0,1077.0,19-07-2024,North
97,T098,C015,Clothing,7.0,643.0,13-08-2023,North
98,T099,C005,Groceries,9.0,429.0,20-01-2024,South


In [13]:
# Display the first 5 rows of the dataset.

df2.head()

,Transaction ID,Customer ID,Product Category,Quantity,Price,Date,Region
0,T001,C015,Books,7.0,167.0,02-02-2023,North
1,T002,C004,Toys,4.0,374.0,17-02-2023,South
2,T003,C033,Furniture,8.0,1059.0,17-05-2024,South
3,T004,C047,Groceries,5.0,304.0,11-02-2024,South
4,T005,C022,Books,7.0,282.0,27-07-2024,West


In [14]:
# Display the last 5 rows.

df2.tail()

,Transaction ID,Customer ID,Product Category,Quantity,Price,Date,Region
95,T096,C036,Furniture,10.0,2901.0,08-08-2023,North
96,T097,C021,Clothing,9.0,1077.0,19-07-2024,North
97,T098,C015,Clothing,7.0,643.0,13-08-2023,North
98,T099,C005,Groceries,9.0,429.0,20-01-2024,South
99,T100,C010,Furniture,8.0,1183.0,07-02-2024,North


In [15]:
# Show the structure of the dataset using info().

df2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 100 entries, 0 to 99
Data columns (total 7 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Transaction ID    100 non-null    object 
 1   Customer ID       100 non-null    object 
 2   Product Category  97 non-null     object 
 3   Quantity          98 non-null     float64
 4   Price             95 non-null     float64
 5   Date              100 non-null    object 
 6   Region            98 non-null     object 
dtypes: float64(2), object(5)
memory usage: 5.6+ KB


In [16]:
# Display the shape of the dataset.

df2.shape

(100, 7)

## **Q3. Column Selection:-**
**Select and display only the following columns:**

- Product

- Region

- Sales

In [45]:
# Ans 3
# As there's no Sales column in the data we have to create one first.
# There are missing values in Quantity and Price column have to fix it.
df2[['Quantity', 'Price']].isnull().sum()

,0
Quantity,2
Price,5


In [108]:
# Rplacing missing values first in Quantity and price column.

df2['Quantity'].fillna(df2['Quantity'].mean(), inplace=True)
df2['Price'].fillna(df2['Price'].mean(), inplace=True)

In [54]:
# Creating new sales column.

df2['Sales'] = np.multiply(df2['Quantity'], df2['Price'])

In [49]:
# displaying columns Product, Region, Sales

df2[['Product Category', 'Region', 'Sales']]

,Product Category,Region,Sales
0,Books,North,1169.0
1,Toys,South,1496.0
2,Furniture,South,8472.0
3,Groceries,South,1520.0
4,Books,West,1974.0
...,...,...,...
95,Furniture,North,29010.0
96,Clothing,North,9693.0
97,Clothing,North,4501.0
98,Groceries,South,3861.0


## **Q4. Indexing using .loc, perform the following:**

- Display the row with index 10.

- Display the Sales value of the row with index 20

In [50]:
# Ans 4
# Displaying the row with index 10.

df2.loc[10]

,10
Transaction ID,T011
Customer ID,C003
Product Category,Electronics
Quantity,4.0
Price,820.526316
Date,04-09-2024
Region,East
Sales,3282.105263


In [109]:
# Displaying the Sales value of the row with index 20.

print(df2.loc[20, 'Sales'])

3074.9387755102043


## **Q5. Indexing using .iloc, display:**

- The first 6 rows and first 4 columns.

- The Sales value from the 3rd row.

In [53]:
# Ans 5
# The first 6 rows and first 4 columns.

df2.iloc[ :6, :4]

,Transaction ID,Customer ID,Product Category,Quantity
0,T001,C015,Books,7.0
1,T002,C004,Toys,4.0
2,T003,C033,Furniture,8.0
3,T004,C047,Groceries,5.0
4,T005,C022,Books,7.0
5,T006,C042,Electronics,10.0


In [73]:
# The Sales value from the 3rd row.

df2.iloc[2:3, 7:8]


,Sales
2,8472.0


## **Q6. Row Slicing**

Display:

- The first 10 rows of the dataset.

- Rows 15 to 25.

In [62]:
# Ans 6
# The first 10 rows of the dataset.

df2.head(10)

,Transaction ID,Customer ID,Product Category,Quantity,Price,Date,Region,Sales
0,T001,C015,Books,7.0,167.0,02-02-2023,North,1169.0
1,T002,C004,Toys,4.0,374.0,17-02-2023,South,1496.0
2,T003,C033,Furniture,8.0,1059.0,17-05-2024,South,8472.0
3,T004,C047,Groceries,5.0,304.0,11-02-2024,South,1520.0
4,T005,C022,Books,7.0,282.0,27-07-2024,West,1974.0
5,T006,C042,Electronics,10.0,866.0,28-12-2024,West,8660.0
6,T007,C022,Furniture,3.0,2340.0,09-04-2023,East,7020.0
7,T008,C023,Clothing,7.0,881.0,14-11-2024,West,6167.0
8,T009,C008,Electronics,8.0,780.0,26-12-2024,East,6240.0
9,T010,C003,Furniture,5.0,1617.0,30-06-2024,North,8085.0


In [70]:
# Rows 15 to 25.

df2.iloc[15:26, : ]

,Transaction ID,Customer ID,Product Category,Quantity,Price,Date,Region,Sales
15,T016,C029,Clothing,5.000000,639.000000,17-05-2024,North,3195.000000
16,T017,C004,Toys,2.000000,423.000000,02-02-2024,South,846.000000
17,T018,C039,Toys,8.000000,820.526316,28-01-2024,NaN,6564.210526
18,T019,C022,Sports,6.000000,1328.000000,26-07-2023,East,7968.000000
19,T020,C045,Toys,2.000000,962.000000,15-01-2023,South,1924.000000
20,T021,C043,Books,5.959184,516.000000,07-07-2024,East,3074.938776
21,T022,C042,Electronics,1.000000,1971.000000,05-04-2024,East,1971.000000
22,T023,C015,Beauty,10.000000,174.000000,21-11-2024,North,1740.000000
23,T024,C012,Clothing,6.000000,236.000000,28-07-2024,West,1416.000000
24,T025,C022,Electronics,9.000000,751.000000,29-08-2023,South,6759.000000


## **Q7. Filtering Data**

**Filter the dataset to display:**

- Rows where Region = "East".

- Rows where Sales > 1500.


In [74]:
# Ans 7
# Rows where Region = "East".

df2[df2['Region'] == 'East']

,Transaction ID,Customer ID,Product Category,Quantity,Price,Date,Region,Sales
6,T007,C022,Furniture,3.000000,2340.000000,09-04-2023,East,7020.000000
8,T009,C008,Electronics,8.000000,780.000000,26-12-2024,East,6240.000000
10,T011,C003,Electronics,4.000000,820.526316,04-09-2024,East,3282.105263
12,T013,C043,NaN,8.000000,510.000000,04-10-2024,East,4080.000000
14,T015,C010,Sports,6.000000,803.000000,06-08-2023,East,4818.000000
18,T019,C022,Sports,6.000000,1328.000000,26-07-2023,East,7968.000000
20,T021,C043,Books,5.959184,516.000000,07-07-2024,East,3074.938776
21,T022,C042,Electronics,1.000000,1971.000000,05-04-2024,East,1971.000000
29,T030,C012,Electronics,4.000000,771.000000,19-08-2023,East,3084.000000
34,T035,C013,Beauty,7.000000,783.000000,08-06-2024,East,5481.000000


In [81]:
# Rows where Sales > 1500.

df2[df2['Sales']> 1500 ]

,Transaction ID,Customer ID,Product Category,Quantity,Price,Date,Region,Sales
2,T003,C033,Furniture,8.0,1059.0,17-05-2024,South,8472.0
3,T004,C047,Groceries,5.0,304.0,11-02-2024,South,1520.0
4,T005,C022,Books,7.0,282.0,27-07-2024,West,1974.0
5,T006,C042,Electronics,10.0,866.0,28-12-2024,West,8660.0
6,T007,C022,Furniture,3.0,2340.0,09-04-2023,East,7020.0
...,...,...,...,...,...,...,...,...
95,T096,C036,Furniture,10.0,2901.0,08-08-2023,North,29010.0
96,T097,C021,Clothing,9.0,1077.0,19-07-2024,North,9693.0
97,T098,C015,Clothing,7.0,643.0,13-08-2023,North,4501.0
98,T099,C005,Groceries,9.0,429.0,20-01-2024,South,3861.0


## **Q8. Multiple Condition Filtering**

**Display rows where:**

- Region is West

- Sales is greater than 1200

In [90]:
# Ans 8

df2[(df2["Region"] == "West") & (df2["Sales"] > 1200)]

,Transaction ID,Customer ID,Product Category,Quantity,Price,Date,Region,Sales
4,T005,C022,Books,7.0,282.0,27-07-2024,West,1974.0
5,T006,C042,Electronics,10.0,866.0,28-12-2024,West,8660.0
7,T008,C023,Clothing,7.0,881.0,14-11-2024,West,6167.0
23,T024,C012,Clothing,6.0,236.0,28-07-2024,West,1416.0
30,T031,C003,Clothing,9.0,938.0,16-10-2024,West,8442.0
37,T038,C011,Books,7.0,648.0,31-08-2023,West,4536.0
42,T043,C028,Clothing,10.0,732.0,27-07-2024,West,7320.0
43,T044,C026,Electronics,9.0,1191.0,07-10-2024,West,10719.0
56,T057,C034,Sports,2.0,800.0,20-01-2024,West,1600.0
57,T058,C026,Electronics,6.0,948.0,12-01-2024,West,5688.0


## **Q9. Descriptive Statistics**

**Generate summary statistics using:**

**describe()**

**From the result identify:**

- Mean sales value

- Maximum quantity sold

- Minimum sales value



In [96]:
# Ans 9

df2.describe()

,Quantity,Price,Sales
count,100.000000,100.000000,100.000000
mean,5.959184,820.526316,4918.348711
std,2.842386,595.536493,4670.183610
min,1.000000,92.000000,285.000000
25%,3.750000,429.000000,1702.000000
50%,6.000000,722.500000,3466.500000
75%,8.000000,969.500000,7336.184211
max,10.000000,2901.000000,29010.000000


**Summary statistics:**

Mean sales value - 4918.348711

Maximum quantity sold - 10

Minimum sales value - 285

##**Q10. Aggregation Analysis**

**Using the Sales column, calculate:**

- Total sales

- Average sales

- Maximum sales

- Minimum sales

In [99]:
# Ans 10
# Total sales

print(df2['Sales'].sum())

491834.8711063373


In [105]:
# Average sales

print(df2['Sales'].mean())

4918.348711063373


In [103]:
# Maximum sales

df2['Sales'].max()

29010.0

In [106]:
# Minimum sales

df2['Sales'].min()

285.0